Modules/Filtering/FFT/src/itkVnlForward1DFFTImageFilterFactory.cxx

```diff
@@ -0,0 +1,32 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include "itkVnlForward1DFFTImageFilter.h"
+#include "itkCreateObjectFunction.h"
+#include "itkVersion.h"
+#include "itkObjectFactoryBase.h"
+
+namespace itk
+{
+// Undocumented API used to register during static initialization.
+// DO NOT CALL DIRECTLY.
+void // ITKFFT_EXPORT
+VnlForward1DFFTFactoryRegister__Private()
```

Comment:
Similar IO classes define an `<name>_EXPORT` macro for factory registration. What is the purpose and what steps need to be taken to create this macro?

Reply:
CMake creates that: https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html

Reply:
Thanks @dzenanz . IO factory files seem to define their own exports manually; see [https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/BioRad/src/itkBioRadImageIOFactory.cxx#L56](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/IO/BioRad/src/itkBioRadImageIOFactory.cxx#L56):
```
// Undocumented API used to register during static initialization.
// DO NOT CALL DIRECTLY.
void ITKIOBioRad_EXPORT
     BioRadImageIOFactoryRegister__Private()
{
  ObjectFactoryBase::RegisterInternalFactoryOnce<BioRadImageIOFactory>();
}
```

Is this intended to be an automated process with CMake? Or can we simply add the module name, i.e. `ITKFFT_EXPORT`?

Reply:
CMake generates the files with the definitions. I think it is `${Project_Name}_EXPORT`. You can look in `${Project_Name}Export.h` to see exactly. So for `ITKFFT` module it should be `ITKFFT_EXPORT`.